### PR TITLE
feat: Join the community

### PR DESCRIPTION
**Pull request by Roxie-32**
The API to join the community

### CODE REVIEW

File: inventorbackend/app/Http/Controllers/JoinUsController.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        $join = JoinUs::create([
+            'email' => $request->email,
+            'track' => $request->track,
+            'portfolio' => ($request->portfolio) ? $request->portfolio : null
+        ]);
+
+        return new JsonResponse(
+            [
+                'success' => 1,
+                'msg' => 'Thank you for indicating your interest to join our community, We will get in touch shortly !'
```
Comment (willypelz) [inline inventorbackend/app/Http/Controllers/JoinUsController.php:24]: can you try and abstract the response for all your API? so you will not have WET code

There are several ways you can abstract the code. 

you can use a class, here is a quick sample using trait. 

```
trait RespondsWithHttpStatus
{
    protected function success($message, $data = [], $status = 200)
    {
        return response([
            'success' => true,
            'data' => $data,
            'message' => $message,
        ], $status);
    }

    protected function failure($message, $status = 422)
    {
        return response([
            'success' => false,
            'message' => $message,
        ], $status);
    }
}
```

Comment (willypelz) [reply inventorbackend/app/Http/Controllers/JoinUsController.php:24]: so you will not have to repeat them all over your code. @Roxie-32

File: inventorbackend/app/Http/Requests/JoinUsRequest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'email' => 'required|email',
```
Comment (willypelz) [inline inventorbackend/app/Http/Requests/JoinUsRequest.php:27]: awesome, this is amazing using form request

File: inventorbackend/database/migrations/2022_11_03_121719_create_join_us_table.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('join_us', function (Blueprint $table) {
+            $table->id();
+            $table->string('email');
```
Comment (willypelz) [inline inventorbackend/database/migrations/2022_11_03_121719_create_join_us_table.php:18]: we can have a duplicate email at the database level. we need to do some level of validation at the migration level. 

unless the business logic supports that.

**Comment by willypelz** [comment]
@Roxie-32, please take a look at the comment on the PR.

**Review by willypelz — CHANGES_REQUESTED** [review]

**Comment by Roxie-32** [comment]
@willypelz kindly check if the response is better now, thanks !

**Pull request merged**